### PR TITLE
Fix problem with namespace

### PR DESCRIPTION
**Pull request by sava-cska**

### CODE REVIEW

**Review by ladisgin — CHANGES_REQUESTED** [review]

File: server/src/types/TypesResolver.cpp [hunk 1]
```diff
@@ -71,14 +71,30 @@ void TypesResolver::resolveStruct(const clang::RecordDecl *D, const std::string
     if (!isCandidateToReplace(id, parent->projectTypes->structs, name)) {
         return;
     }
+
+    std::string currentStructName = canonicalType.getNonReferenceType().getUnqualifiedType().getAsString();
```
[the remark this answers is not in the document]
Comment (sava-cska) [reply server/src/types/TypesResolver.cpp:75]: Ok

File: server/src/types/TypesResolver.cpp [hunk 2]
```diff
@@ -182,8 +198,25 @@ void TypesResolver::resolveEnum(const clang::EnumDecl *EN, const string &name) {
     if (!isCandidateToReplace(id, parent->projectTypes->enums, name)) {
         return;
     }
+
+    std::string currentEnumName = canonicalType.getNonReferenceType().getUnqualifiedType().getAsString();
```
[the remark this answers is not in the document]
Comment (sava-cska) [reply server/src/types/TypesResolver.cpp:202]: Ok

File: server/src/types/TypesResolver.cpp [hunk 3]
```diff
@@ -71,14 +71,30 @@ void TypesResolver::resolveStruct(const clang::RecordDecl *D, const std::string
...
+            LOG_S(WARNING) << StringUtils::stringFormat("Type of struct %s doesn't start with 'struct'", currentStructName);
+        } else {
+            currentStructName = currentStructName.substr(7);
+        }
+    }
+    fullname.insert(std::make_pair(currentStructName, currentStructName));
+    if (const clang::RecordDecl *parentNode = llvm::dyn_cast<const clang::RecordDecl>(D->getLexicalParent())) {
+        std::string parentRecordName = parentNode->getNameAsString();
+        if (!parentRecordName.empty()) {
+            fullname[currentStructName] = fullname[parentRecordName] + "::" + currentStructName;
```
[the remark this answers is not in the document]
Comment (sava-cska) [reply server/src/types/TypesResolver.cpp:87]: No

File: server/src/types/TypesResolver.cpp [hunk 4]
```diff
@@ -71,14 +71,30 @@ void TypesResolver::resolveStruct(const clang::RecordDecl *D, const std::string
...
+
+    std::string currentStructName = canonicalType.getNonReferenceType().getUnqualifiedType().getAsString();
+    if (!currentStructName.empty()) {
+        if (!StringUtils::startsWith(currentStructName, "struct ")) {
+            LOG_S(WARNING) << StringUtils::stringFormat("Type of struct %s doesn't start with 'struct'", currentStructName);
+        } else {
+            currentStructName = currentStructName.substr(7);
+        }
+    }
+    fullname.insert(std::make_pair(currentStructName, currentStructName));
```
[the remark this answers is not in the document]
Comment (sava-cska) [reply server/src/types/TypesResolver.cpp:83]: All will be ok

**Review by ladisgin — APPROVED** [review]

**Pull request merged**